text/trace/0174-http-semantic-conventions.md

```diff
@@ -0,0 +1,186 @@
+which will serve as a basis for [stabilizing](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#stable)
+the [existing semantic conventions for HTTP](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md),
+which are currently in an [experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#experimental)
+state. The goal is to declare HTTP semantic conventions stable before the
+end of Q1 2022.
+
+## Motivation
+
+Most observability scenarios involve HTTP communication. For Distributed Tracing
+to be useful across the entire scenario, having good observability for
+HTTP is critical. To achieve this, OpenTelemetry must provide stable conventions
+and guidelines for instrumenting HTTP communication.
+
+Bringing the existing experimental semantic conventions for HTTP to a
+stable state is a crucial step for users and instrumentation authors, as it
+allows them to rely on [stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability),
+and thus to ship and use stable instrumentation.
+
+> NOTE. This OTEP captures a scope for changes should be done to existing
+experimental semantic conventions for HTTP, but does not propose solutions.
+
+## Roadmap for v1.0
+
+1. This OTEP, consisting of scenarios/open questions and a proposed roadmap, is
+   approved and merged.
+2. [Stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability)
+   for semantic conventions are approved and merged. This is not strictly related
+   to semantic conventions for HTTP but is a prerequisite for stabilizing any
+   semantic conventions.
+3. Separate PRs addressing the scenarios and open questions listed in this
+   document are approved and merged.
+4. Proposed specification changes are verified by prototypes for the scenarios
+   and examples below.
+5. The [specification for HTTP semantic conventions for tracing](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md)
+   are updated according to this OTEP and are declared
+   [stable](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#stable).
+
+The steps in the roadmap don't necessarily need to happen in the given order,
+some steps can be worked on in parallel.
+
+## Scope for v1.0: scenarios and open questions
+
+> NOTE. The scope defined here is subject for discussions and can be adjusted
+  until this OTEP is merged.
+
+Scenarios and open questions mentioned below must be addressed via separate PRs.
+
+### Error status defaults
```

Comment:
This part seems kind of prescriptive too. Is there somewhere that this is documented as the accepted decision? I think you could make a case for client 4xx spans being UNSET or ERROR. I agree that UNSET is probably better due to the edge cases. Either way, we need to get consensus in the community, has there been any prior discussion on this topic?

Reply:
Ping @jamesmoessis @denisivan0v

Reply:
I don't think this conversation should block merging, so if it's ready go ahead and merge. I will raise PR in the spec regarding error statuses and we can discuss there.

Reply:
@jamesmoessis that's correct, we need to get consensus on this in the community - that's the reason we have this item in scope. Currently, the spec states:

> For HTTP status codes in the 4xx range span status MUST be left unset in case of SpanKind.SERVER and MUST be set to Error in case of SpanKind.CLIENT.

However as you pointed out, there are some edge cases when it would be better to keep status as UNSET (e.g. 409 CONFLICT just indicates that data was modified by other client - which is not a client error; the same could be applicable to 429 or even 404 in case a resource not yet created).

I'd suggest to keep this item and discuss this topic on the next Instrumentation SIG meeting.

Does this help? Please let me know if it works for you and this discussion can be resolved.

Reply:
@jamesmoessis you added your comment while I was typing mine :) Looks, we are on the same page here, resolving the discussion.